lib/cloud_controller/opi/stager_client.rb

```diff
@@ -9,8 +9,22 @@ module OPI
   class StagerClient < BaseClient
     def stage(staging_guid, staging_details)
       logger.info('stage.request', staging_guid: staging_guid)
-      staging_request = to_request(staging_guid, staging_details)
 
+      if staging_details.lifecycle.type == VCAP::CloudController::Lifecycles::DOCKER
+        complete_staging(staging_guid, staging_details)
+      elsif staging_details.lifecycle.type == VCAP::CloudController::Lifecycles::BUILDPACK
+        staging_request = to_request(staging_guid, staging_details)
+        start_staging(staging_guid, staging_request)
+      else
+        raise("lifecycle type `#{staging_details.lifecycle.type}` is invalid")
```

Comment:
nit: usually we'd define an Error type scoped to the class itself, like this. 

https://github.com/cloudfoundry/cloud_controller_ng/blob/32f180a4d6ffcc9302d79dc2cb1fe74814fd9c54/app/actions/deployment_cancel.rb#L3-L4